lib/util/common.js

```diff
@@ -12,10 +12,23 @@ function addHook (fastify, pluginOptions) {
   const sharedSchemasMap = new Map()
 
   fastify.addHook('onRoute', (routeOptions) => {
-    // we need to skip HEAD route
-    // since it may automatically added by fastify
-    // and will conflict with the existing route
-    if (routeOptions.method === 'HEAD') return
+    // we need to skip the HEAD route if there is
+    // already one route with the same operationId.
+    // since https://github.com/fastify/fastify/pull/2826 head routes
+    // are created by default in fastify
+    if (
+      routeOptions.method === 'HEAD' &&
+      routes.find(
```

Comment:
It depends on the route registration order which will be wrong in future.
I would just add a option to force expose head route here.

It can be placed inside `schema` using `x-` extension or `route options` with a proper name.

Reply:
So like this:
```
routeOptions.method === 'HEAD' && routeOptions.schema['x-force-include-head'] !== undefined
```
or so
```
routeOptions.method === 'HEAD' && routeOptions.forceIncludeHead === true
```
?

The second one would require changes in fastify, right?

Reply:
> The second one would require changes in fastify, right?

No, it do not require changes in `fastify`.

Reply:
Ah nice 
The second one seems cleaner imo.
Please tell me which do you prefer and if the name `forceIncludeHead` suits you then I change the PR 🙂